### PR TITLE
Use `.is_some_and()` instead of `.is_some() && .unwrap()`

### PR DESCRIPTION
**Pull request by WaffleLapkin**
That's nicer & more readable.

(I just noticed that this looks weird while trying to understand why zed changes my cursor, so decided to make a quick fix (btw the issue with the cursor is that zed always loads cursor named "default" on wayland))

Release Notes:

- N/A

### CODE REVIEW

File: crates/gpui/src/platform/linux/platform.rs [hunk 1]
```diff
@@ -74,7 +74,7 @@ impl Default for LinuxPlatform {
 impl LinuxPlatform {
     pub(crate) fn new() -> Self {
         let wayland_display = env::var_os("WAYLAND_DISPLAY");
-        let use_wayland = wayland_display.is_some() && !wayland_display.unwrap().is_empty();
+        let use_wayland = wayland_display.is_some_and(|dp| !dp.is_empty());
```
Comment (maxdeviant) [inline crates/gpui/src/platform/linux/platform.rs:77]: Can we use a more meaningful binding name here?

```suggestion
        let use_wayland = wayland_display.is_some_and(|display| !display.is_empty());
```

Comment (WaffleLapkin) [reply crates/gpui/src/platform/linux/platform.rs:77]: I don't think it's _necessary_ (the full `wayland_display` is right here anyway), but I don't really have a preference.

**Comment by WaffleLapkin** [comment]
@cla-bot check

File: crates/gpui/src/platform/linux/platform.rs (hunk 1 above)
Comment (maxdeviant) [reply crates/gpui/src/platform/linux/platform.rs:77]: I think using `display` here would be better, just for clarity's sake.

**Review by maxdeviant — APPROVED** [review]
Thanks!

**Pull request merged**